pkg/application/consul/consul.go

```diff
@@ -0,0 +1,82 @@
+package consul
+
+import (
+	"github.com/awslabs/eksdemo/pkg/application"
+	"github.com/awslabs/eksdemo/pkg/cmd"
+	"github.com/awslabs/eksdemo/pkg/installer"
+	"github.com/awslabs/eksdemo/pkg/template"
+)
+
+// https://developer.hashicorp.com/consul/docs/k8s/installation/install
+// https://developer.hashicorp.com/consul/tutorials/kubernetes/kubernetes-eks-aws
+// https://artifacthub.io/packages/helm/hashicorp/consul
+
+func NewApp() *application.Application {
+	app := &application.Application{
+		Command: cmd.Command{
+			Name:        "consul",
+			Description: "HashiCorp Consul Service-Mesh",
+		},
+
+		Installer: &installer.HelmInstaller{
+			ChartName:     "consul",
+			ReleaseName:   "consul",
+			RepositoryURL: "https://helm.releases.hashicorp.com",
+			VersionField:  "1.4.1",
+			ValuesTemplate: &template.TextTemplate{
+				Template: valuesTemplate,
+			},
+		},
+	}
+
+	app.Options, app.Flags = newOptions()
+
+	return app
+
+}
+
+const valuesTemplate = `---
+global:
+  # The main enabled/disabled setting.
+  # If true, servers, clients, Consul DNS and the Consul UI will be enabled.
+  # Namespace
+  namespace: {{ .Namespace }}
+  # The prefix used for all resources created in the Helm chart.
+  name: {{ .Namespace }}
+  # The name of the datacenter that the agents should register as.
+  datacenter: {{ .Datacenter }}
```

Comment:
can you add `--datacenter` flag so this is configurable for the user?

Reply:
Added flag